code/mob/dead/observer.dm

```diff
@@ -101,14 +101,45 @@
 	if (!P.AH)
 		return
 
+	var/is_mutantrace = FALSE
+	if ("lizard" in P.traitPreferences.traits_selected)
+		src.icon_state = "ghost-lizard"
+		is_mutantrace = TRUE
+	else if ("cow" in P.traitPreferences.traits_selected)
+		src.icon_state = "ghost-cow"
+		is_mutantrace = TRUE
+	else if ("skeleton" in P.traitPreferences.traits_selected)
+		src.icon_state = "ghost-skeleton"
+		is_mutantrace = TRUE
+	else if ("roach" in P.traitPreferences.traits_selected)
+		src.icon_state = "ghost-roach"
+		is_mutantrace = TRUE
+	else if ("pug" in P.traitPreferences.traits_selected)
+		src.icon_state = "ghost-pug"
+		is_mutantrace = TRUE
+
+	var/mutantrace_hair = is_mutantrace && ("bald" in P.traitPreferences.traits_selected)
```

Comment:
This should also check for the hidden mutantrace hair admin option ideally.

Reply:
You'll have to let me know what variable that is, I don't know anything about admin vars x_x

Reply:
`hair_override`

Reply:
If you are a /mob/new_player when this proc is started, hair_override var won't be accessible?

Reply:
Then just handle that case, it can only be added after they've joined anyway.